packages/vite/src/node/build.ts

```diff
@@ -451,7 +451,7 @@ export function resolveBuildEnvironmentOptions(
   }
 
   if (resolved.cssMinify == null) {
-    resolved.cssMinify = !!resolved.minify
+    resolved.cssMinify = raw.ssr ? 'esbuild' : !!resolved.minify
```

Comment:
I'm using `raw.ssr` to align with `minify: raw.ssr ? false : 'esbuild',` above. Do we want to update that to `consumer === 'server'`? I guess we don't want to, because we would want to enable minify for edge runtimes.

Reply:
I don't quite understand the question, would edge runtimes be `consumer == 'server'` true too? Checking `consumer` seems correct to me.

Reply:
Yes, that is correct. The comments were talking about `minify` option (not the `cssMinify` option). It was suggesting to change `minify: raw.ssr ? false : 'esbuild'` to `minify: consumer === 'server' ? false : 'esbuild'` or `minify: consumer === 'server' && !notWorker ? false : 'esbuild'`. Why this is related to this PR is because the condition should be same between `minify` and `cssMinify`. For example, if minify is changed to `minify: consumer === 'server' && !notWorker ? false : 'esbuild'`, then cssMinify should be `cssMinify: consumer === 'server' && !notWorker ? 'esbuild' : !!resolved.minify`. It is because we want to set a different default value from `!!resolved.minify` only when `resolved.minify` was implicitly set to `false` (= when the default value was used).

Reply:
Ah everything makes sense now, I didn't realize you we're talking about changing to `consumer === 'server'` to the `minify` option.

My first intuition is that I agree with you at "I guess we don't want to, because we would want to enable minify for edge runtimes.", but then we're partial towards edge runtime being extended, e.g. what if it's a rsc environment.

`notWorker` check seems to be a better addition check to alleviate this case, would it be based on `webCompatible`? I guess my only fear with all these later is that it could be hard to explain in docs.

Would be good to have @patak-dev chime in too as this relates to fallback values for different environments.

Reply:
Yes, we missed this one. We should move to `minify: consumer === 'server' ? false : 'esbuild'` (for both).

About edge environments, they can set `minify: false` as their default:
```
function edgeEnvironment(userConfig) {
  return mergeConfig({
      build: {
        minify: false
      }
      dev: {
        createEnvironment: ...
      }
    },
    userConfig
  );
}
```

Reply:
Replaced with `consumer === 'server'` 👍